### PR TITLE
CLOUDP-212065: fix: add 2.0.0 major version of AKO

### PR DESCRIPTION
**Pull request by s-urbaniak**
## Proposed changes

Starting from the changes present in this PR, rendering AKO v1.x resources is not supported any more. Starting from here, rendering AKO v2.x resources is possible only.

_Jira ticket:_ CLOUDP-210683

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

**Comment by s-urbaniak** [comment]
@mongodb/docs-cloud-team please be aware of the changes mentioned in the PR title. Please let me know if you have further questions, thank you!

File: test/e2e/atlas/helper_test.go
Comment (gssbzn) [inline test/e2e/atlas/helper_test.go:273]:
```suggestion
```

File: go.mod
Comment (gssbzn) [inline go.mod:61]: this lib is missing an owner

File: test/e2e/atlas/operator_helper_test.go
Comment (gssbzn) [inline test/e2e/atlas/operator_helper_test.go:161]: we have our own that I'd like to use instead of more dependencies with kube
```suggestion
	deployment.Spec.Replicas = pointer.Get(0)
```

File: test/e2e/atlas/kubernetes_config_apply_test.go
Comment (gssbzn) [inline test/e2e/atlas/kubernetes_config_apply_test.go:87]: is this flag now mandatory or was it added here for the test

Comment (s-urbaniak) [reply test/e2e/atlas/kubernetes_config_apply_test.go:87]: it was added as concurrently executing tasks stomped on each other in the same namespace.

File: test/e2e/atlas/kubernetes_config_generate_test.go
Comment (s-urbaniak) [inline test/e2e/atlas/kubernetes_config_generate_test.go:1136]: note to myself: check if this is still executed, this should use generated names.

File: internal/kubernetes/operator/secrets/secrets.go
Comment (igor-karpukhin) [inline internal/kubernetes/operator/secrets/secrets.go:37]: Interesting approach :) This can also be implemented with a struct, eliminating the `Build()` method at the end.

Comment (s-urbaniak) [reply internal/kubernetes/operator/secrets/secrets.go:37]: Indeed, it was something real quick, I don't think we need anything fancy here. If we're going to need more initialising variables we can also introduce a struct. The `Build` method is just to denote that this is a builder so its implementation can vary independently.

File: test/e2e/atlas/operator_helper_test.go
Comment (igor-karpukhin) [inline test/e2e/atlas/operator_helper_test.go:161]: This has to be `int32` not `int`

**Review by igor-karpukhin — COMMENTED** [review]
There is one issue with the type check. LGTM in general, if the CI is green.

File: test/e2e/atlas/operator_helper_test.go
Comment (s-urbaniak) [reply test/e2e/atlas/operator_helper_test.go:161]: @gssbzn fyi ^ probably that's why the other method was used.

Comment (gssbzn) [reply test/e2e/atlas/operator_helper_test.go:161]: pointer.Get is a generic
```
pointer.Get[int32](0)
```

File: internal/kubernetes/operator/features/crds.go
Comment (gssbzn) [inline internal/kubernetes/operator/features/crds.go:50]: [q] we are dropping support for v1 then? I'm ok just confirming

File: internal/kubernetes/operator/features/crds_test.go
Comment (gssbzn) [reply internal/kubernetes/operator/features/crds_test.go:278]: the original -1 was on the patch, which currently was 2.0.2, so it would produce 2.-1.2, my -1 is on major, producing 1.0.2 ;)

File: test/e2e/atlas/operator_helper_test.go
Comment (s-urbaniak) [reply test/e2e/atlas/operator_helper_test.go]: fixed

File: internal/kubernetes/operator/features/crds_test.go
Comment (s-urbaniak) [reply internal/kubernetes/operator/features/crds_test.go:278]: fixed

**Comment by s-urbaniak** [comment]
@gssbzn i hopefully addressed all your concerns, thanks for the feedback, PTAL. Having said that I prefixed this PR as WIP (work in progress). We unfortunately found another bug yesterday in AKO v2.0 which we are fixing. The implied changes here most likely will be:
- We will not advertise `2.0.0` in the Atlas-CLI, just the `2.0.1` release including the bug fixes
- reference the `2.0.1` images

cc @igor-karpukhin

File: go.mod
Comment (s-urbaniak) [reply go.mod:61]: fixed

**Review by gssbzn — DISMISSED** [review]
Changes LGTM you can decide when is best to merge this once the operator fixes are out but I'm not blocking the PR

**Review by igor-karpukhin — DISMISSED** [review]
Great work! 👍

**Comment by s-urbaniak** [comment]
before we merge, I'd like to implement one last optimization. Currently, we always generate empty secret templates. That's not really necessary unless a concrete alert integration is configured in Atlas.

We can consider the integration type name to see if we actually need to create a secret template.

**Review by igor-karpukhin — DISMISSED** [review]
Looks good 👍

**Review by igor-karpukhin — APPROVED** [review]

**Pull request merged**
